tools/sklearn/search_model_validation.xml

```diff
@@ -551,7 +667,118 @@ The evaluation of settings supports operations in Math, list comprehension, nump
+max_iter=10, mean_change_tol=0.001, n_components=10, n_topics=None, perp_tol=0.1,
+random_state=0, topic_word_prior=None, total_samples=1000000.0, verbose=0)
+
+20:  sklearn_decomposition.MiniBatchDictionaryLearning(alpha=1, batch_size=3, dict_init=None, fit_algorithm='lars',
+n_components=None, n_iter=1000, random_state=0, shuffle=True, split_sign=False,
+transform_algorithm='omp', transform_alpha=None, transform_n_nonzero_coefs=None, verbose=False)
+
+21:  sklearn_decomposition.MiniBatchSparsePCA(alpha=1, batch_size=3, callback=None, method='lars', n_components=None,
+n_iter=100, random_state=0, ridge_alpha=0.01, shuffle=True, verbose=False)
+
+22:  sklearn_decomposition.NMF(alpha=0.0, beta_loss='frobenius', init=None, l1_ratio=0.0, max_iter=200,
+n_components=None, random_state=0, shuffle=False, solver='cd', tol=0.0001, verbose=0)
+
+23:  sklearn_decomposition.PCA(copy=True, iterated_power='auto', n_components=None, random_state=0, svd_solver='auto', tol=0.0, whiten=False)
+
+24:  sklearn_decomposition.SparsePCA(U_init=None, V_init=None, alpha=1, max_iter=1000, method='lars', n_components=None,
+random_state=0, ridge_alpha=0.01, tol=1e-08, verbose=False)
+
+25:  sklearn_decomposition.TruncatedSVD(algorithm='randomized', n_components=2, n_iter=5, random_state=0, tol=0.0)
+
+26:  sklearn_kernel_approximation.Nystroem(coef0=None, degree=None, gamma=None, kernel='rbf', kernel_params=None, n_components=100, random_state=0)
+
+27:  sklearn_kernel_approximation.RBFSampler(gamma=1.0, n_components=100, random_state=0)
+
+28:  sklearn_kernel_approximation.AdditiveChi2Sampler(sample_interval=None, sample_steps=2)
+
+29:  sklearn_kernel_approximation.SkewedChi2Sampler(n_components=100, random_state=0, skewedness=1.0)
+
+30:  sklearn_cluster.FeatureAgglomeration(affinity='euclidean', compute_full_tree='auto', connectivity=None,
+linkage='ward', memory=None, n_clusters=2, pooling_func=<function mean at 0x113078ae8>)
+
+31:  skrebate_ReliefF(discrete_threshold=10, n_features_to_select=10, n_neighbors=100, verbose=False)
+
+32:  skrebate_SURF(discrete_threshold=10, n_features_to_select=10, verbose=False)
+
+33:  skrebate_SURFstar(discrete_threshold=10, n_features_to_select=10, verbose=False)
+
+34:  skrebate_MultiSURF(discrete_threshold=10, n_features_to_select=10, verbose=False)
+
+35:  skrebate_MultiSURFstar(discrete_threshold=10, n_features_to_select=10, verbose=False)
+
+'sk_prep_all': All sklearn preprocessing estimators, i.e., 0-7
+
+'fs_all': All feature_selection estimators, i.e., 8-14
+
+'decomp_all': All decomposition estimators, i.e., 15-25
+
+'k_appr_all': All kernel_approximation estimators, i.e. 26-29
+
+'reb_all': All skrebate estimators, i.e. 31-35
+
+'imb_all': All imbalanced-learn sampling methods, i.e. 36-54. CAUTION: Mix of imblearn and other preprocessors may not work.
+
+'all': All, 0-36
+
+Support mix (except imblearn methods), e.g.:
+
+- : [None, 'sk_prep_all', 22, 'k_appr_all', sklearn_feature_selection.SelectKBest(k=50)]
```

Comment:
the `- :` is part of the parameter or not? Why do we not offer these 35 options in a select field? Or in a repeat?

Reply:
The `-` is not part of the parameter, but cheetah markdown. The point in your second question is very good. If we do that, it will change the UI dramatically though. Let me think about it.